File: src/main/java/de/maggiwuerze/xdccloader/util/State.java
package de.maggiwuerze.xdccloader.util;

public enum State {

    CONNECTING("Connecting"),
    DONE("Done"),
    ERROR("Error : '%s'"),
    FINALIZING("Finalizing"),
    PREPARING("Preparing"),
    PREPARED("Prepared"),
    READY("Ready"),
    RESTARTING("Restarting"),
    STOPPED("Stopped"),
    TRANSMITTING("Transmitting"),
    UNKNOWN("Unknown");

    String externalString;

    State(String externalString) {

        this.externalString = externalString;
    }

    public String getExternalString() {
        return externalString;
    }
}

File: src/main/kotlin/de/maggiwuerze/xdccloader/util/State.kt
package de.maggiwuerze.xdccloader.util

enum class State(val externalString : String) {

    PREPARING("Preparing"),
    PREPARED("Prepared"),
    READY("Ready"),
    CONNECTING("Connecting"),
    TRANSMITTING("Transmitting"),
    FINALIZING("Finalizing"),
    DONE("Done"),
    ERROR("Error code: '%s'"),
    UNKNOWN("Unknown")
}